### PR TITLE
GEODE-3247 Document query method invocation changes

### PR DESCRIPTION
**Pull request by karensmolermiller**
@dihardman @joeymcallister @davebarnes97 @jhuynh1 @metatype @upthewaterspout Please review.

### CODE REVIEW

File: geode-docs/developing/query_select/the_where_clause.html.md.erb [hunk 1]
```diff
@@ -235,6 +235,35 @@ If more than one method can be invoked, the query processor chooses the method w
...
+`replaceFirst`, `split`, `startsWith`, `substring`, `toCharArray`, 
+`toLowerCase`, `toUpperCase`, or `trim` on a `String` object
+- `intValue`, `longValue`, `shortValue`, etc. on any `Number` object
+- `booleanValue` on any `Boolean` object
+- `equals`, `compareTo`, or `toString` on any object
+
+To disable the authorization check,
+start all servers with the added system property
+`QueryService.allowUntrustedMethodInvocation`.
+For example,
```
Comment (joeymcallister) [inline geode-docs/developing/query_select/the_where_clause.html.md.erb]: Change comma to a colon in "For example:"

File: geode-docs/developing/query_select/the_where_clause.html.md.erb [hunk 2]
```diff
@@ -235,6 +235,35 @@ If more than one method can be invoked, the query processor chooses the method w
 
 The query processor uses the runtime types of the parameters and the receiver to determine the proper method to invoke. Because runtime types are used, an argument with a null value has no typing information, and so can be matched with any object type parameter. When a null argument is used, if the query processor cannot determine the proper method to invoke based on the non-null arguments, it throws an `AmbiguousNameException`.
 
+**Methods calls with the `SecurityManager` enabled**
+
+When the `SecurityManager` is enabled,
+<%=vars.product_name%> throws a `NotAuthorizedException` when any
+method not on this list of whitelisted methods is invoked:
+
+- `keySet`, `entrySet`, `values`, `containsKey` or `get` on a `Map`, `Collection`, or `Region` object
```
Comment (joeymcallister) [inline geode-docs/developing/query_select/the_where_clause.html.md.erb]: Might it be clearer to begin each of these list items with the object type they refer to? That is, would a reader want to begin with the object type and then work down to the methods? E.g.:
- **On a `Region.Entry` object:** `getKey` or `getValue`
- **On a `Date` or `Timestamp` object:** `after`, `before`, `getNanos`, or `getTime`

File: geode-docs/developing/query_select/the_where_clause.html.md.erb [hunk 3]
```diff
@@ -235,6 +235,35 @@ If more than one method can be invoked, the query processor chooses the method w
...
+- `getKey` or `getValue`  on a `Region.Entry` object
+- `after`, `before`, `getNanos`, or `getTime` on a `Date` or `Timestamp` object
+- `charAt`, `codePointAt`, `codePointBefore`, `codePointCount`, 
+`compareToIgnoreCase`, `concat`, `contains`, `contentEquals`, 
+`endsWith`, `equalsIgnoreCase`, `getBytes`, `hashCode`, `indexOf`, 
+`intern`, `isEmpty`, `lastIndexOf`, `length`, `matches`, 
+`offsetByCodePoints`, `regionMatches`, `replace`, `replaceAll`, 
+`replaceFirst`, `split`, `startsWith`, `substring`, `toCharArray`, 
+`toLowerCase`, `toUpperCase`, or `trim` on a `String` object
```
Comment (upthewaterspout) [inline geode-docs/developing/query_select/the_where_clause.html.md.erb]: Maybe this should just be "any method on a String"

File: geode-docs/developing/query_select/the_where_clause.html.md.erb [hunk 4]
```diff
@@ -235,6 +235,35 @@ If more than one method can be invoked, the query processor chooses the method w
...
+- `booleanValue` on any `Boolean` object
+- `equals`, `compareTo`, or `toString` on any object
+
+To disable the authorization check,
+start all servers with the added system property
+`QueryService.allowUntrustedMethodInvocation`.
+For example,
+
+```
+gfsh>start server --name=Server1 --J=QueryService.allowUntrustedMethodInvocation
```
Comment (upthewaterspout) [inline geode-docs/developing/query_select/the_where_clause.html.md.erb]: I think this should be -J=-DQueryService.allowUntrustedMethodInvocation

Comment (davebarnes97) [reply geode-docs/developing/query_select/the_where_clause.html.md.erb]: Two hyphens before the "J" is correct in a gfsh command. I think Dan just meant to add the "-D".

Comment (karensmolermiller) [reply geode-docs/developing/query_select/the_where_clause.html.md.erb]: Good catch.  Thanks.  In fact, I think that the option should read
`--J=-Dgemfire.QueryService.allowUntrustedMethodInvocation=true`

File: geode-docs/developing/query_select/the_where_clause.html.md.erb [hunk 5]
```diff
@@ -241,27 +241,22 @@ When the `SecurityManager` is enabled,
...
-`offsetByCodePoints`, `regionMatches`, `replace`, `replaceAll`, 
-`replaceFirst`, `split`, `startsWith`, `substring`, `toCharArray`, 
-`toLowerCase`, `toUpperCase`, or `trim` on a `String` object
-- `intValue`, `longValue`, `shortValue`, etc. on any `Number` object
-- `booleanValue` on any `Boolean` object
-- `equals`, `compareTo`, or `toString` on any object
+- On a `Map`, `Collection`, or `Region` object: `keySet`, `entrySet`, `values`, `containsKey` or `get`
+- On a `Region.Entry` object: `getKey` or `getValue`
+- On a `Date` or `Timestamp` object: `after`, `before`, `getNanos`, or `getTime`
+- Any method on a `String` object
```
Comment (joeymcallister) [inline geode-docs/developing/query_select/the_where_clause.html.md.erb]: Change this to "On a `String` object: any method" for consistency.

**Comment by dihardman** [comment]
+1 with latest changes.

**Pull request merged**